src/reducers/sortListReducer.js

```diff
@@ -0,0 +1,21 @@
+import { SET_SORT } from '../actions/actionTypes';
+
+export default (state = {}, action) => {
+  const {
+    key,
+    field,
+    direction,
+  } = action;
+  switch (action.type) {
+    case SET_SORT:
+      return {
+        ...state,
```

Comment:
Do you really want to keep the previous state? All the sorting component does is changes the sorting type. So, your state should change completely.

Reply:
Yes, i really want to keep the previous state.
The component saves fort direction for all columns:
```
sortListReducer: {
    'DashboardId:Column': {
      field: 'name',
      direction: 'DESC'
    },
    'DashboardId:Column': {
      field: 'priority',
      direction: 'ASC'
    },
    'DashboardId:Column': {
      field: 'name',
      direction: 'ASC'
    },
}
```
So, if we change the state of one of columns, we should keep state of others.